### PR TITLE
DO NOT SQUASH: Add more analytical modal tests (3)

### PR DESCRIPTION
**Pull request by ruansava**
This pull request introduces several improvements and refactorings to the analytical modal solver and error management utilities, as well as associated tests. The main changes include enhancements to the analytical solver for ellipsoidal domains, improved handling of Dirichlet boundary conditions, a refactor of a utility function for cleaning numerical arrays, and test updates to reflect these changes.

**Analytical Modal Solver Improvements:**

* Replaced the use of `broyden1` with more robust and widely used root-finding and minimization routines (`fsolve`, `minimize_scalar`) for finding roots in the analytical modal solver, improving reliability and accuracy for both 2D/3D circular and elliptical/ellipsoidal cases. Also, implemented a Rayleigh-Ritz approach for the 3D ellipsoidal Dirichlet case.
* Improved the handling of Dirichlet boundary conditions in the hypershape frequency factor calculation by introducing a multiplicative factor (`fbc_dirichlet`) to adjust the frequency factors for Dirichlet versus Neumann conditions, ensuring more accurate analytical results.

**Code Quality and Consistency:**
* Refactored the function `clean_inst_num` to `sanitize_num_array` for clarity and updated all references and tests accordingly. 
* Improved test enum naming and usage for clarity and to avoid confusion in error manager tests.

**Bug Fixes and Minor Cleanups:**
* Fixed integration measure initialization in `c_equivalent` to ensure default quadrature is correctly handled.
* Fixed a bug in the calculation of `fr_rec` for 3D cases in the fit function.

**Documentation Updates:**
* Updated module and docstrings in tests to reflect the new analytical solver focus and clarified descriptions.

### CODE REVIEW

File: spyro/solvers/eikonal/eikonal_eq.py [hunk 1]
```diff
@@ -584,7 +584,7 @@ def nonlinear_solution(self, wf_parameters, nl_solver='vinewtonssls',
                 c, c_min, V, diam_mesh).dat.data_with_halos[:]
         else:
             # Clean numerical instabilities
-            data_eikL = clean_inst_num(lin_sol.dat.data_with_halos[:])
+            data_eikL = sanitize_num_array(lin_sol.dat.data_with_halos[:])
```
Comment (Olender) [inline spyro/solvers/eikonal/eikonal_eq.py:587]: Thanks!

File: spyro/solvers/modal/modal_ana_sol.py [hunk 2]
```diff
@@ -550,7 +592,7 @@ def c_equivalent(self, c, V, quad_rule=None,
                              "Function", none_default=True)
 
         # Integration measure
-        dx = fire_dx(**quad_rule) if quad_rule else fire_dx
+        dx = fire_dx(**quad_rule) if quad_rule else fire_dx(**{"degree": 0})
```
Comment (Olender) [inline spyro/solvers/modal/modal_ana_sol.py:595]: WHat does this degree do?

Comment (ruansava) [reply spyro/solvers/modal/modal_ana_sol.py:595]: Polynomial degree of numerical integration. Basically, a constant variation is considered across the interval. I tested higher degrees and got the same results for the tests.

**Review by Olender — APPROVED** [review]

**Review by Olender — CHANGES_REQUESTED** [review]
one minor change (theres an added character)

File: tests/on_one_core/test_modal_analytical_solver.py [hunk 3]
```diff
@@ -480,7 +479,7 @@ def test_modal(wave_instance, element_geometry, dimension, degree_layer, homogen
      0.02  69.442
      0.03  70.974
      0.04  73.179
-     0.05  75.766
+     0.05  7¡5.766
```
Comment (Olender) [inline tests/on_one_core/test_modal_analytical_solver.py]:
```suggestion
     0.05  75.766
```

**Review by Olender — APPROVED** [review]
